server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -133,7 +133,7 @@
   private static final int DELETE_SIZE = 50;
   private static final String ERROR_PARSING_SQL =
       "meet error while parsing SQL to physical plan: {}";
-  public static Vector<SqlArgument> sqlArgumentsList = new Vector<>();
+  public static final Vector<SqlArgument> sqlArgumentsList = new Vector<>();
```

Comment:
why not use `CopyOnWriteArrayList`?

Reply:
After reading codes, I find `sqlArgumentsList` is for MetricService, which is for logging the query history (and its execution plan..). So, the field is write-intense and read-rare. 
Is CopyOnWriteArrayList has better performance than `synchronized ` in this way?

Besides, @qiaojialin, don't you need to add lock for reading the list? (line 46 in QueryServlet.java)

Reply:
Hi, the sqlArgumentList is a write-intensive with nearly no read. I plan to use the Collections.synchronizedList to avoid synchronization everywhere.